src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt

```diff
@@ -15,28 +15,19 @@ fun CommandEvent.arguments(): Arguments {
         , args.filter { !it.isFlag }, args.filter { it.isFlag })
 }
 
-suspend fun Message.getMentionedMember(index: Int, action: suspend (Member?) -> Unit = {}) {
-    val str = contentRaw.split(WHITESPACE_REGEX).getOrNull(index) ?: let {
-        action(null)
-        return
-    }
+suspend fun Message.getMentionedMember(index: Int): Member? {
+    val str = contentRaw.split(WHITESPACE_REGEX).getOrNull(index) ?: return null
     val id = str.toLongOrNull()
     if (id != null) {
-        action(guild.retrieveMemberById(id.toString(), false).await { cont, _ ->
+        return (guild.retrieveMemberById(id.toString(), false).await { cont, _ ->
             cont.resume(null)
```

Comment:
why is this needed?

Reply:
wdym? in case the user isn't cached

Reply:
i mean the new await function - can't you just do `return guild.retrieveMemberById(id.toString(), false).await()`?

Reply:
in case it errors due to no user found... 
https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/entities/Guild.html#retrieveMemberById(long)
If the user is invalid, it errors

Reply:
ah makes sense, maybe a try / catch would be nicer but this seems alright

Reply:
> @dkim19375 it would seem that this doesnt compile

that's why i said don't merge :))

Reply:
why did it reply here